featuretools/primitives/standard/transform_primitive.py

```diff
@@ -906,3 +907,138 @@ def lag(time_index, numeric):
+            lats = (self.lats[0] <= transposed[0]) & \
+                   (self.lats[1] >= transposed[0])
+            longs = (self.lons[0] <= transposed[1]) & \
+                    (self.lons[1] >= transposed[1])
+            return lats & longs
+        return geobox
+
+
+class GeoMidpoint(TransformPrimitive):
+    """Determines the geographic center of two coordinates.
+
+    Examples:
+        >>> geomidpoint = GeoMidpoint()
+        >>> geomidpoint([(42.4, -71.1)], [(40.0, -122.4)])
+        [(41.2, -96.75)]
+    """
+    name = "geomidpoint"
+    input_types = [ColumnSchema(logical_type=LatLong), ColumnSchema(logical_type=LatLong)]
+    return_type = ColumnSchema(logical_type=LatLong)
+    commutative = True
+
+    def get_function(self):
+        def geomidpoint_func(latlong_1, latlong_2):
+            lat_1s = np.array([x[0] if isinstance(x, tuple) else np.nan for x in latlong_1])
+            lon_1s = np.array([x[1] if isinstance(x, tuple) else np.nan for x in latlong_1])
+            lat_2s = np.array([x[0] if isinstance(x, tuple) else np.nan for x in latlong_2])
+            lon_2s = np.array([x[1] if isinstance(x, tuple) else np.nan for x in latlong_2])
+            lat_middle = np.array([lat_1s, lat_2s]).transpose().mean(axis=1)
+            lon_middle = np.array([lon_1s, lon_2s]).transpose().mean(axis=1)
+            return list(zip(lat_middle, lon_middle))
+        return geomidpoint_func
+
+
+class CityblockDistance(TransformPrimitive):
+    """Calculates the distance between points in a city road grid.
+
+    Description:
+        This distance is calculated using the haversine formula, which
+        takes into account the curvature of the Earth.
+        If either input data contains `NaN`s, the calculated
+        distance with be `NaN`.
+        This calculation is also known as the Mahnattan distance.
+
+    Args:
+        unit (str): Determines the unit value to output. Could
+            be miles or kilometers. Default is miles.
+
+    Examples:
+        >>> cityblock_distance = CityblockDistance()
+        >>> DC = (38, -77)
+        >>> Boston = (43, -71)
+        >>> NYC = (40, -74)
+        >>> distances_mi = cityblock_distance([DC, DC], [NYC, Boston])
+        >>> np.round(distances_mi, 3).tolist()
+        [301.519, 672.089]
+
+        We can also change the units in which the distance is calculated.
+
+        >>> cityblock_distance_kilometers = CityblockDistance(unit='kilometers')
+        >>> distances_km = cityblock_distance_kilometers([DC, DC], [NYC, Boston])
+        >>> np.round(distances_km, 3).tolist()
+        [485.248, 1081.622]
+    """
+    name = "cityblock_distance"
+    input_types = [ColumnSchema(logical_type=LatLong), ColumnSchema(logical_type=LatLong)]
+    return_type = ColumnSchema(logical_type=Double, semantic_tags={'numeric'})
+    commutative = True
+
+    def __init__(self, unit='miles'):
+        if unit not in ['miles', 'kilometers']:
+            raise ValueError("Invalid unit given")
```

Comment:
Probably not a big deal, but should we add a simple test to cover this?

Reply:
Done.
https://github.com/alteryx/featuretools/pull/1814/commits/cfd3ed455daf44315986f912172b4df9880dc79d